### ISSUE TITLE
EfficientPose performance

### ISSUE THREAD

**Issue by wbenoot** [daniegr/EfficientPose#14]
I was wondering, when I use different EfficientPose models on my own data the results are not really great. Your documentation clearly says "_Avoid that the subject is occluded by other objects, even partial occlusion is discouraged_" but the paper of EfficientPose promises very high results on the MPII Dataset, a dataset where occlusions are quite common. Why do the actual results of these pretrained models not live up to the results of the paper / how can I get the promissed performance? Thank you in advance!

**Comment by daniegr** [comment]
@wbenoot: Thank you for showing interest in our project! The recommendations of usage is mainly to obtain maximal performance, but as you mention by being trained on the diverse MPII dataset the models should in itself be able to combat some occlusion. Regarding the issue you are experiencing, I would be interested in getting some more details about the way you are preprocessing the images that you are feeding into the model. I

**Comment by wbenoot** [comment]
I use an EfficientDet Human Detector and send those crops to the EfficientPose model. I've noticed that adding padding around the crops actually increases the performance a bit, but compared to an older pose estimator like AlphaPose the results are not particularly better. Is there any specific type of preprocessing I should do?

**Comment by wbenoot** [comment]
I should note that I of course also use the same preprocessing done in the source code (resizing to right resolution, imagenet preprocessing function...)

**Comment by daniegr** [comment]
@wbenoot: Would you mind sharing some examples from your tests, in particular of cases where you are not satisfied with your results? And have you tried exploring the different EfficientPose models?

**Comment by wbenoot** [comment]
I uploaded some different pictures comparing an older version of AlphaPose to different EfficientPose models. We clearly see that AP has a very good result and I was hoping one of the EP models would achieve a similar result (especially since only the RT and I variant are faster). When I use these models on pictures of a lower quality with more occlusions or harder poses (horizontal or crouched) I just cannot seem to produce usable outputs... 

[AlphaPose](https://imgur.com/QsgWYEk)
[EfficientPoseIII](https://imgur.com/esoeaUF)
[EfficientPoseRT](https://imgur.com/CFvxB8w)

**Comment by daniegr** [comment]
@wbenoot: I would suggest that you experiment a bit with the appropriate padding/distance to the person. You may try out the live demo for this purpose.

**Comment by wbenoot** [comment]
Alright, and there is no possibility of something in the weights being wrong?

**Comment by wbenoot** [comment]
@daniegr I've played around with different paddings on the human detection crops and noticed that with the right padding the poses of even the RT model can be really decent! The only problem I still have is that there doesn't seem to be a "one correct padding size/method", sometimes 50% results better than 20% and vica versa.

**Comment by daniegr** [comment]
@wbenoot: For your information, when we train the models we apply the padding scheme of [Xiao et al.](https://github.com/leoxiaobin/deep-high-resolution-net.pytorch/blob/master/lib/dataset/mpii.py), which is about 25% of padding relative to the approximate scale of the person.

**Comment by chaowentao** [comment]
@wbenoot Hello, can you share with me the model training details? I also want to implement the model. Thank you very much.

**Comment by daniegr** [comment]
@chaowentao: Most of the details are available in our paper (https://link.springer.com/article/10.1007/s10489-020-01918-7). However, if you have specific questions feel free to reach out.
